source/deployment/elasticsearch.rst

```diff
@@ -0,0 +1,78 @@
+Elasticsearch Beta (E20)
+========================
+
+*Available in Enterprise Edition E20.*
+
+`Elasticsearch <https://www.elastic.co/guide/en/elasticsearch/reference/current/index.html>`_ is a distributed, RESTful search engine capable of supporting search in large scale enterprise deployments with millions of messages. 
+
+.. toctree::
+    :maxdepth: 2
+
+Deployment Guide
+----------------
+
+Overview
+~~~~~~~~
+
+Elasticsearch allows you to search large volumes of data quickly, and in near real time, by creating and managing an index of post data. The indexing process can be managed from the System Console after setting up and connecting an Elasticsearch server. The post index is stored on the Elasticsearch server and is updated constantly after new posts are made. In order to index existing posts, a bulk index of the enitre post database must be generated. 
+
+Setting up an Elasticsearch Server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The setup process for the Elasticsearch server is documented in the `official Elasticsearch documentation <https://www.elastic.co/guide/en/elasticsearch/reference/current/setup.html>`_. 
+
+XXXXXX George: Any special steps specific to Mattermost implementation not mentioned in the offical docs? 
+
+Hardware Requirements
+`````````````````````
+Elasticsearch should run alone on a server and use all of the resources available to it.
+
+XXXXXX George: hardware requirements or recommendations for the elasticsearch server?
```

Comment:
Don't really have specific recommendations at the moment - mostly use your judgement and Elasticsearch best practices. I guess for production they should run a proper Elasticsearch cluster, whatever that is.